yt_dlp/extractor/redbee.py

```diff
@@ -269,8 +272,17 @@ def _real_extract(self, url):
         embed_page = self._download_webpage(
             'https://www.rtbf.be/auvio/embed/' + ('direct' if live else 'media'),
             media_id, query={'id': media_id})
-        data = self._parse_json(self._html_search_regex(
-            r'data-media="([^"]+)"', embed_page, 'media data'), media_id)
+
+        media_data = self._html_search_regex(r'data-media="([^"]+)"', embed_page, 'media data', fatal=False)
+        if not media_data:
+            if re.findall(r'<div[^>]+id="[^"]*js-error-expired[^"]*"[^>]+class="(?![^"]*hidden)', embed_page):
+                raise ExtractorError('Livestream has ended.', expected=True)
+            if re.findall(r'<div[^>]+id="[^"]*js-sso-connect[^"]*"[^>]+class="(?![^"]*hidden)', embed_page):
+                self.raise_login_required()
```

Comment:
```suggestion
            if re.search(r'<div[^>]+id="[^"]*\bjs-error-expired\b', embed_page):
                raise ExtractorError('Livestream has ended', expected=True)
            if re.search(r'<div[^>]+id="[^"]*\bs-sso-connect\b', embed_page):
                self.raise_login_required()
```
is this insufficient?

Reply:
Nop, all errors are already inside the page and they remove the 'hidden' class to show it

```html
<div id="js-error-geoloc" class="rtbf-embed-message hidden">
        <figure class="rtbf-embed-message__placeholder">
            <img src="[https://ds1.static.rtbf.be/media/program/image/ng_d41a344a2252c5891b77-324x183@2x.jpg](view-source:https://ds1.static.rtbf.be/media/program/image/ng_d41a344a2252c5891b77-324x183@2x.jpg)" alt="Les minuscules" />
            <div class="www-full-stretch www-pattern"></div>
            <figcaption class="rtbf-embed-message__body">
                <h1>Les minuscules</h1>
                <h2></h2>
                <p>
                    Ce contenu est visible uniquement en Belgique<br>
                    En Belgique ?  Consultez notre
                    <a class="rtbf-embed-message__link" href="[https://www.rtbf.be/auvio/faq](view-source:https://www.rtbf.be/auvio/faq)" target="_top">FAQ</a>
                </p>
                ....
```

Reply:
gotcha. 
* replace `re.findall` with `re.search`
* Can you explain `id="[^"]*js-error-expired[^"]*"`? Shouldn't it be checking just `id="js-error-expired"`?

Reply:
I was thinking that the regex had to be flexible, I'll change it n-n